ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/data/Model.kt

```diff
@@ -113,6 +114,45 @@ data class Model(
 
     var optionalInfo: Map<String, String> = mapOf()
 
+    private fun getFormat(list: List<String>): String {
+        return list.let {
+            if (it.isNotEmpty()) {
+                it.joinToString(",")
+            } else {
+                "static"
+            }
+        }
+    }
+
+    private fun getTensors(list: List<String>?, format: String): String {
+        return if (format == "static") {
+            val numTensors = list?.size ?: 1
+            ",num_tensors=$numTensors"
+        } else {
+            ""
+        }
+    }
+
+    private fun getType(list: List<String>): String {
+        return list.let {
+            if (it.isNotEmpty()) {
+                ",types=${it.joinToString(",")}"
+            } else {
+                ""
+            }
+        }
```

Comment:
```kotlin
fun getType(list: List<String>): String {
    return list.let {
        if (it.isNotEmpty()) {
            ",types=${it.joinToString(",")}"
        } else {
            ""
        }
    }
}

fun main() {
    val stringList = listOf("Type1", "Type2", "Type3")
    val stringListEmpty = listOf("Type1", "", "Type3")
    val stringListBlank = listOf("Type1", " ", "Type3")
    val stringListAllEmpty = listOf<String>()
    
    println("stringList: ${getType(stringList)}")
    println("stringListEmpty: ${getType(stringListEmpty)}")
    println("stringListBlank: ${getType(stringListBlank)}")
    println("stringListAllEmpty: ${getType(stringListAllEmpty)}")
    
    val filtered = stringListEmpty.filterNot{ it.isEmpty() or it.isBlank() }
    val types = if (filtered.isEmpty()) {
    	""   
    } else {
        ",types=${filtered.joinToString(",")}"
    }

    println("$types")
}
```
```bash
stringList: ,types=Type1,Type2,Type3
stringListEmpty: ,types=Type1,,Type3
stringListBlank: ,types=Type1, ,Type3
stringListAllEmpty: 
,types=Type1,Type3
```

```suggestion
        val filtered = stringListEmpty.filterNot{ it.isEmpty() or it.isBlank() }
        return if (filtered.isEmpty()) {
            ""
        } else {
            ",types=${filtered.joinToString(",")}"
        }
```

or 

```suggestion
        return stringListEmpty.filterNot{ it.isEmpty() or it.isBlank() }.let {
            if (it.isEmpty()) {
               ""
            } else {
              ",types=${filtered.joinToString(",")}"
            }
        }
```
However, I am not sure that ```let``` is really required in this case.

Reply:
This is the reason why we need unit test 😭
Thank you for the detailed review
Updated!